6ogo/data-viz-modeling
# Add UI improvements and data visualization components

This PR adds several UI improvements and data visualization capabilities:

1. Data Visualization Components:
   - LineChart for trend analysis
   - BarChart for value distribution
   - ScatterPlot for correlation analysis

2. Features:
   - Responsive chart components using Recharts
   - Card-based layout for better organization
   - Sample visualization page

3. Component Features:
   - Customizable titles and descriptions
   - Responsive container sizing
   - Tooltips and grid lines
   - Flexible data key mapping

These components provide a foundation for visualizing uploaded data. Each chart component is reusable and can be configured with different data sources and keys.

To test:
1. Run the development server
2. Navigate to /visualizations
3. Check that all charts render properly
4. Verify responsive behavior
5. Test tooltips and interactions
